src/dominion/player.rs

```diff
@@ -54,14 +74,213 @@ impl Player {
             self.shuffle_deck();
         }
 
+        // TODO handle empty deck
```

Comment:
/Isn't it so that if the pile is empty (after shuffling the discard pile) then nothing happens? If so, simply adding the check `if !self.deck_pile.is_empty() {` should be enough,

But perhaps the caller might wanna know if a draw actually happened. Then I guess returning a `bool` makes sense?

Reply:
Yeah I guess we'll figure out how to do this later. `remove` will panic if the desk is empty.